src/module.c

```diff
@@ -1693,6 +1695,9 @@ void SpecialCaseCtx_Free(specialCaseCtx* ctx) {
   if (!ctx) return;
   if(ctx->specialCaseType == SPECIAL_CASE_KNN) {
     QueryNode_Free(ctx->knn.queryNode);
+    if (ctx->knn.pq) {
+      heap_destroy(ctx->knn.pq);
```

Comment:
unrelated to the core of this PR, potential memory leak fix

Reply:
Although searchRequestCtx_Free->SpecialCaseCtx_Free is called in rscParseRequest, the heap is only created in `searchResultReducer` (later in the query flow) and also released there:

```
  if (rCtx.pq) {
    heap_destroy(rCtx.pq);
  }
  if (rCtx.reduceSpecialCaseCtxKnn &&
      rCtx.reduceSpecialCaseCtxKnn->knn.pq) {
    heap_destroy(rCtx.reduceSpecialCaseCtxKnn->knn.pq);
    rCtx.reduceSpecialCaseCtxKnn->knn.pq = NULL;
  }
```
I am removing this to prevent coverage hits.